perllib/FixMyStreet/Cobrand/Bromley.pm

```diff
@@ -1146,6 +1153,255 @@ sub garden_waste_cost {
+                        name => 'uprn',
+                        value => $uprn,
+                    },
+                    {
+                        name => 'Subscription_Details_Container_Type',
+                        value => 44
+                    },
+                    {
+                        name => 'Subscription_Details_Quantity',
+                        value => $self->waste_get_sub_quantity($service),
+                    },
+                );
+                $renew->set_extra_metadata('dd_date', $date);
+                $renew->insert;
+                $handled = 1;
+            }
+        } elsif ( $category eq 'Cancel' ) {
+            next unless $payment->{Status} eq 'Processed';
+            $rs = $rs->search({ category => { -in => ['Garden Subscription', 'Cancel Garden Subscription'] } });
+            my $p;
+            while ( my $cur = $rs->next ) {
+                my $sub_type = $cur->get_extra_field_value('Subscription_Type');
+                if ( $sub_type eq $self->waste_subscription_types->{New} ) {
+                    $p = $cur;
+                } elsif ( $cur->category eq 'Cancel Garden Subscription' ) {
+                    # already processed
+                    next RECORD if $cur->get_extra_metadata('dd_date') && $cur->get_extra_metadata('dd_date') eq $date;
+                }
+            }
+            if ( $p ) {
+                my $service = $self->waste_get_current_garden_sub( $p->get_extra_field_value('property_id') );
+                unless ($service) {
+                    warn "no matching service to cancel for $reference\n";
+                    next;
+                }
+                my $cancel = _duplicate_waste_report($p, 'Cancel Garden Subscription');
+                $cancel->set_extra_fields(
+                    {
+                        name => 'service_id',
+                        value => 545,
+                    },
+                    {
+                        name => 'uprn',
+                        value => $uprn,
+                    },
+                    {
+                        name => 'Container_Instruction_Action',
+                        value => $self->waste_container_actions->{remove},
+                    },
+                    {
+                        name => 'Container_Instruction_Container_Type',
+                        value => 44
+                    },
+                    {
+                        name => 'Container_Instruction_Quantity',
+                        value => $self->waste_get_sub_quantity($service),
+                    },
+                );
+                $cancel->set_extra_metadata('dd_date', $date);
+                $cancel->insert;
+                $handled = 1;
+            }
+        } else {
+            next unless $payment->{Status} eq 'Paid';
+            # we fetch the confirmed ones as well as we explicitly want to check for
+            # processed reports so we can warn on those we are missing.
```

Comment:
I don't see how this comment is implemented - either a next RECORD line is hit, or handled is set to 1, I can't see how it will ever get to the unhandled warning.